src/styles/app-select/AppSelect.tsx

```diff
@@ -0,0 +1,204 @@
+      handleToggleList();
+    }
+  };
+
+  const handleSelect = (item: { value: string; disabled?: boolean }) => {
+    if (item.disabled) {
+      return;
+    }
+
+    setSelected(prevSelected => {
+      let updatedSelected: string[];
+
+      if (type === "multiple") {
+        updatedSelected = prevSelected.includes(item.value)
+          ? prevSelected.filter(option => option !== item.value)
+          : [...prevSelected, item.value];
+      } else if (type === "single") {
+        updatedSelected = prevSelected.includes(item.value) ? [] : [item.value];
+      } else {
+        updatedSelected = prevSelected;
+      }
+
+      if (onChange) {
+        onChange(updatedSelected);
+      }
+      return updatedSelected;
+    });
+  };
+
+  const isSelected = (item: { value: string }) => selected.includes(item.value);
+
+  const handleScroll = async ({
+    scrollOffset,
+    scrollHeight,
+  }: {
+    scrollOffset: number;
+    scrollHeight: number;
+  }) => {
+    if (scrollOffset + Number(height) >= scrollHeight && hasMore && !loading) {
+      setLoading(true);
+      await loadMoreOptions?.();
+      setLoading(false);
+    }
+  };
+
+  return (
+    <div className="app-select" ref={ref}>
+      <div
+        style={{ width }}
+        role="button"
+        tabIndex={0}
+        className="app-select__button"
+        onClick={handleToggleList}
+        onKeyUp={handleToggleWithKeyboard}
+        aria-expanded={isOpen}
+        aria-label={isOpen ? "Collapse content" : "Expand content"}
+        aria-haspopup="listbox"
+      >
+        <p>{label}</p>
+        <Icon
+          className="app-select_arrow-button"
+          icon="tabler:chevron-down"
+          width="24"
+          height="24"
+        />
+      </div>
+
+      <FixedSizeList
+        ref={listRef}
+        className={classNames("app-select__list", {
+          "app-select--expanded": isOpen,
+        })}
+        height={height}
+        width={width}
+        itemCount={options.length + (hasMore ? 1 : 0)}
+        itemSize={itemSize}
+        itemData={{
+          options,
+          isSelected,
+          onSelect: handleSelect,
+          renderOptions,
+          loading,
+          selectedAdornment,
+        }}
+        overscanCount={overscanCount}
+        onScroll={({ scrollOffset }) => {
+          const list = listRef.current;
+          if (list) {
+            handleScroll({
+              scrollOffset,
+              scrollHeight: list.props.itemSize * list.props.itemCount,
+            });
+          }
+        }}
+      >
+        {Item}
+      </FixedSizeList>
+    </div>
+  );
+};
+
+const Item = memo(({ index, style, data }: ItemProps) => {
+  const {
+    options,
+    isSelected,
+    onSelect,
+    renderOptions,
+    loading,
+    selectedAdornment,
+  } = data;
+
+  if (index === options.length) {
+    return (
+      <div className="app-select__spinner" style={style}>
+        {loading && <AppSpinner />}
+      </div>
```

Comment:
I think we should not render `app-select__spinner` when loading is happening

Reply:
In my opinion, user won't understand that the options are being fetched if there is no spinner

Reply:
No, i meant following:
```suggestion
      {loading && <div className="app-select__spinner" style={style}>
         <AppSpinner />
      </div>}
```

Reply:
Yeah, fixed